src/matrix/kernelkroneckermat.jl

```diff
@@ -25,3 +25,37 @@ end
     k(x,x') = ∏ᵢᴰ k(xᵢ,x'ᵢ)
 """
 @inline iskroncompatible(κ::Kernel) = false # Default return for kernels
+
+@inline ismatrixkroncompatible(κ::MOKernel) = false # Default return for kernels
+@inline ismatrixkroncompatible(κ::IndependentMOKernel) = true
+@inline ismatrixkroncompatible(κ::IntrinsicCoregionMOKernel) = true
+
+function _mo_kernelmatrix_kron(::MOInputIsotopicByFeatures, K, B)
+    return Kronecker.kronecker(K, B)
+end
+
+function _mo_kernelmatrix_kron(::MOInputIsotopicByOutputs, K, B)
+    return Kronecker.kronecker(B, K)
+end
+
+function kernelkronmat(
+    k::IndependentMOKernel, x::MOI, y::MOI
+) where {MOI<:IsotopicMOInputsUnion}
+    @assert x.out_dim == y.out_dim
+    Ktmp = kernelmatrix(k.kernel, x.x, y.x)
+    mtype = eltype(Ktmp)
+    return _mo_kernelmatrix_kron(Ktmp, Eye{mtype}(x.out_dim), x)
+end
+
+function kernelkronmat(
+    k::IntrinsicCoregionMOKernel, x::MOI, y::MOI
+) where {MOI<:IsotopicMOInputsUnion}
+    @assert x.out_dim == y.out_dim
+    Ktmp = kernelmatrix(k.kernel, x.x, y.x)
+    return _mo_kernelmatrix_kron(Ktmp, k.B, x)
+end
+
+function kernelkronmat(k::MOK, x::MOI) where {MOI<:IsotopicMOInputsUnion,MOK<:MOKernel}
```

Comment:
```suggestion
function kernelkronmat(k::MOKernel, x::IsotopicMOInputsUnion)
```
I believe the short-form syntax should be fine here, no?

Reply:
I am never quite sure when one or the other is needed, or if they are actually the same.

Reply:
In the context of method definitions, the only difference is some compiler heuristics around when or when not to specialise, for a very particular set of types: https://docs.julialang.org/en/v1/manual/performance-tips/#Be-aware-of-when-Julia-avoids-specializing . For the most part, these heuristics are there for a good reason (compiler latency), so it's usually only worth overriding them in special cases. In this instance, I don't believe there's any difference whatsoever between the two.

The place where it matters most of the time is inside `struct` declarations.